src/main/java/net/imglib2/algorithm/localextrema/LocalExtrema.java

```diff
@@ -182,32 +183,72 @@
 			final ExecutorService service,
 			final int numTasks )
 	{
+		final long[] borderSize = getRequiredBorderSize( shape, source.numDimensions() );
+		return findLocalExtrema( source, shrink( source, borderSize ), localNeighborhoodCheck, shape, service, numTasks );
+	}
+
+	/**
+	 * Find pixels that are extrema in their local neighborhood. The specific
+	 * test for being an extremum can be specified as an implementation of the
+	 * {@link LocalNeighborhoodCheck} interface.
+	 *
+	 * The task is parallelized along the longest dimension of
+	 * <code>interval</code>
+	 *
+	 * @param source
+	 *            Find local extrema of the function defined by this
+	 *            {@link RandomAccessible}
+	 * @param interval
+	 *            Domain in which to look for local extrema. It is the callers
+	 *            responsibility to ensure that <code>source</code> is defined
+	 *            in all neighborhoods of <code>interval</code>.
+	 * @param localNeighborhoodCheck
+	 *            Check if current pixel qualifies as local maximum. It is the
+	 *            callers responsibility to pass a
+	 *            {@link LocalNeighborhoodCheck} that avoids the center pixel if
+	 *            <code>shape</code> does not skip the center pixel.
+	 * @param shape
+	 *            Defines the local neighborhood.
+	 * @param service
+	 *            {@link ExecutorService} handles parallel tasks
+	 * @param numTasks
+	 *            Number of tasks for parallel execution
+	 * @return {@link ArrayList} of extrema
+	 */
+	public static < P, T > ArrayList< P > findLocalExtrema(
```

Comment:
I personally prefer to be as generic as possible with the input parameters and as specific as possible with the return values. The caller can still write `List< P > extrema = findLocalExtrema(...)`. Returning `List` would, of course, allow to change the implementation later on. This is, however, not an interface but a (set of) static method and the implementation details are unlikely to change.